crates/holochain/src/conductor/conductor.rs

```diff
@@ -866,30 +866,39 @@ mod network_impls {
         /// Get signed agent info from the conductor
         pub async fn get_agent_infos(
             &self,
-            _cell_id: Option<CellId>,
-        ) -> ConductorApiResult<Vec<AgentInfoSigned>> {
-            unimplemented!()
-            /*
+            cell_id: Option<CellId>,
+        ) -> ConductorApiResult<Vec<Arc<AgentInfoSigned>>> {
             match cell_id {
                 Some(c) => {
-                    let (d, a) = c.into_dna_and_agent();
-                    let db = self.p2p_agents_db(&d);
-                    Ok(get_single_agent_info(db.into(), d, a)
+                    let (dna_hash, agent_key) = c.into_dna_and_agent();
+                    let peer_store = self
+                        .holochain_p2p
+                        .peer_store(dna_hash)
+                        .await
+                        .map_err(|err| ConductorApiError::CellError(err.into()))?;
+                    Ok(peer_store
+                        .get(agent_key.to_k2_agent())
                         .await?
-                        .map(|a| vec![a])
+                        .map(|agent_info| vec![agent_info])
                         .unwrap_or_default())
                 }
                 None => {
+                    let dna_hashes = self
+                        .spaces
+                        .get_from_spaces(|space| (*space.dna_hash).clone());
                     let mut out = Vec::new();
-                    // collecting so the mutex lock can close
-                    let envs = self.spaces.get_from_spaces(|s| s.p2p_agents_db.clone());
-                    for db in envs {
-                        out.append(&mut all_agent_infos(db.into()).await?);
+                    for dna_hash in dna_hashes {
```

Comment:
I was trying this out with parallel futures all joined, but it was so messy with `Vec<Result<Vec<` and the error handling, that I preferred this call. It won't have to stand high performance scenarios, I believe.